### PR TITLE
Adjustments to ec_deployment provisioning

### PR DESCRIPTION
**Pull request by tnthornton**
### Description of your changes
In its current state, provider-jet-ec does not effectively support creating `Deployment`s in ESS. This changeset adds the following functionality:
- extending supported credentials to apiKey
- adjusted ec_deployment config to allow for async deploys and associating provider provisioned id
- piped ESS connection details through to the MRs connection details secret
- reduced resources currently produced to ec_deployment as that is currently the only functional resource

In addition two examples are added:
1. A minimal sample deployment.
2. A more extensive example that also provisions Kibana and APM as part of the deployment.

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
I've verified that the included examples:
* properly create their respective deployments
* the deployments successfully sync along with their corresponding connection details
* deleting the MR results in the EC resource being deleted

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: config/ec/deployment.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"fmt"
+
+	"github.com/crossplane/terrajet/pkg/config"
+)
+
+// Configure configures individual resources by adding custom
+// ResourceConfigurators.
+func Configure(p *config.Provider) {
```
Comment (hasheddan) [inline config/ec/deployment.go:22]: might be worth pulling some of the values below out into `const` in the future 👍🏻

File: internal/clients/ec.go [hunk 2]
```diff
@@ -91,11 +96,42 @@ func TerraformSetupBuilder(version, providerSource, providerVersion string) terr
...
+	// set environment variables for sensitive provider configuration
+	// NOTE(@tnthornton) either apiKey OR username and password are valid
+	// not both.
+	apikey := ecCreds[keyAPIKey]
+	uname := ecCreds[keyUsername]
+	pword := ecCreds[keyPassword]
+
+	if apikey != "" && (uname != "" || pword != "") {
+		return nil, errors.New(errEitherAPIKeyOrUP)
+	}
```
Comment (hasheddan) [inline internal/clients/ec.go:121]: Is this check redundant with the subsequent two? If purpose is to inform that api key or username / pass can be used I think this error can just be moved to the uname / pass check below after the api key check

**Review by hasheddan — APPROVED** [review]
@tnthornton two small nitpicks below but non-blocking, thanks!

File: config/ec/deployment.go (hunk 1 above)
Comment (tnthornton) [reply config/ec/deployment.go:22]: Oh ya, good call out 👍

File: internal/clients/ec.go (hunk 2 above)
Comment (tnthornton) [reply internal/clients/ec.go:121]: That's fair. We can move the intent to just being informational. The checks are currently there to guard against:
* API Key defined
* Username defined

Technically we'll end up using the API Key, but it's still not a "valid" config.

**Pull request merged**